apps/api/src/services/form.service.ts

```diff
@@ -23,6 +23,72 @@ export class FormService {
 		return await FormModel.getAllFormsWithSubmissionsCount(eventId);
 	}
 
+	async deleteFormField(formId: string, fieldId: string) {
+		try {
+			const formFieldInfo = await FormFieldsModel.findOne({
+				id: fieldId,
+				fk_form_id: formId,
+			});
+			if (!formFieldInfo) {
+				throw FFError.notFound(
+					`${SystemTable.FormFields}: Query Failed : 
+          form with id ${formId} could not be found`,
+				);
+			}
+
+			await FormFieldsModel.delete({
+				id: fieldId,
+			});
+
+			return {
+				ok: true,
+				message: 'Field deleted successfully',
+			};
+		} catch {
+			throw new InternalServerErrorException();
+		}
+	}
+
+	async editFormField(data: EditFormFieldDto, formId: string, fieldId: string) {
+		try {
+			const formInfo = await FormModel.findOne({
+				id: formId,
+			});
+			if (!formInfo) {
+				throw FFError.notFound(
+					`${SystemTable.FormFields}: Query Failed : 
+          form with id ${formId} could not be found`,
+				);
+			}
+
+			const existingField = await FormFieldsModel.findOne({
+				fk_form_id: fieldId,
+				id: fieldId,
+			});
+
+			await FormFieldsModel.update(
+				{
+					id: fieldId,
+				},
+				{
+					placeholder: data.placeholder || existingField?.placeholder,
+					name: data.label?.length ? data.label : existingField?.name,
+					required: Object.hasOwn(data, 'require')
+						? data.require
+						: existingField?.required,
+					type: data.type || existingField?.type,
+				},
+			);
```

Comment:
Avoid doing this, Only include the fields which needs to be updated

Reply:
this  , we dont know which field is updating right ?  that is why I am doing this , any other ways ?

Reply:
You can just pass the data here. No need for explicit mapping here